lib/Doctrine/DBAL/Connection.php

```diff
@@ -993,20 +993,33 @@ public function getTransactionNestingLevel() : int
     }
 
     /**
-     * Returns the ID of the last inserted row, or the last value from a sequence object,
-     * depending on the underlying driver.
+     * Returns the ID of the last inserted row.
      *
-     * Note: This method may not return a meaningful or consistent result across different drivers,
-     * because the underlying database may not even support the notion of AUTO_INCREMENT/IDENTITY
-     * columns or sequences.
-     *
-     * @param string|null $name Name of the sequence object from which the ID should be returned.
+     * If the underlying driver does not support identity columns, an exception is thrown.
      *
      * @return string A string representation of the last inserted ID.
+     *
+     * @throws Driver\DriverException
+     */
+    public function lastInsertId() : string
+    {
+        return $this->getWrappedConnection()->lastInsertId();
+    }
+
+    /**
+     * Returns the current sequence value for the given sequence name.
+     *
+     * If the underlying driver does not support sequences, an exception is thrown.
+     *
+     * @param string $name The sequence name.
+     *
+     * @return string The current sequence value.
```

Comment:
I wonder why we should cast the sequence value to a string and can't return it as is (untyped in the API) given that it's always a number.

Reply:
I guess that this boils down to 32-bit PHP being still supported, when identity columns / sequences may return 64-bit integers, which can only be accurately represented as strings.

Reply:
Correct. And this is up to the underlying DB driver which I don't think we should interfere with.

Reply:
So you don't want to use strong typing here? (unless you're willing to target PHP 8.0, in which case you will be able to use [union types](https://wiki.php.net/rfc/union_types_v2) `int|string`).

Reply:
> So you don't want to use strong typing here?

No, I don't. It doesn't seem to make sense at the abstraction level.

Reply:
Can't say I'm convinced on this one, but anyway: fixed in 18d245b7b2cdebbd0632175bb9eaa2117b9d173d.

Reply:
BTW, found some more arguments on the topic from a year ago: https://github.com/doctrine/dbal/pull/3074#issuecomment-382549778.

Reply:
BTW, found some more arguments on the topic from a year ago: https://github.com/doctrine/dbal/pull/3074#issuecomment-382549778.